### PR TITLE
Limit subscriptions for a single connection

### LINKED ISSUE #1042: Add a limit for the number of subscriptions from a single connection
Add an env var that controls the maximum allowed subscriptions from a single connection. When the limit is exceeded no subscription should be created and an appropriate error message should be sent to the client.

### PR DESCRIPTION
**Pull request by Ivo-Yankov**
**Description**:
Adds an env variable that controls the maximum allowed number of active subscriptions for a single connection.

**Related issue(s)**:

Fixes #1042 

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -37,3 +37,4 @@ WEB_SOCKET_PORT=
 CONNECTION_LIMIT=
 WS_MAX_CONNECTION_TTL=
 WS_CONNECTION_LIMIT_PER_IP=
+WS_SUBSCRIPTION_LIMIT=
```
Comment (AlfredoG87) [inline .env.example:40]: Description for this ENV property should be included on [Docs](https://github.com/hashgraph/hedera-json-rpc-relay/blob/main/docs/configuration.md)

File: packages/relay/src/lib/errors/JsonRpcError.ts [hunk 2]
```diff
@@ -206,5 +206,10 @@ export const predefined = {
     name: 'Could not retrieve latest block',
     code: -32607,
     message: 'Error encountered retrieving latest block'
+  }),
+  'MAX_SUBSCRIPTIONS': new JsonRpcError({
+    name: 'Exceeding maximum allowed subscriptions',
```
Comment (Nana-EC) [inline packages/relay/src/lib/errors/JsonRpcError.ts:211]: nit:
```suggestion
    name: 'Exceeded maximum allowed subscriptions',
```

File: packages/relay/src/lib/errors/JsonRpcError.ts [hunk 3]
```diff
@@ -206,5 +206,10 @@ export const predefined = {
...
+    code: -32608,
+    message: 'Exceeding maximum allowed subscriptions'
```
Comment (Nana-EC) [inline packages/relay/src/lib/errors/JsonRpcError.ts:213]: nit:
```suggestion
    message: 'Exceeded maximum allowed subscriptions'
```

File: packages/relay/src/lib/subscriptionController.ts [hunk 4]
```diff
@@ -138,4 +140,4 @@ export class SubscriptionController {
             });
         }
     }
-}
+}
```
Comment (Nana-EC) [inline packages/relay/src/lib/subscriptionController.ts:143]: nit: restore new line

File: packages/relay/tests/lib/subscriptionController.spec.ts [hunk 5]
```diff
@@ -216,4 +216,4 @@ describe("subscriptionController", async function() {
         expect(subId).to.be.eq(subId2);
     });
 
-});
+});
```
Comment (Nana-EC) [inline packages/relay/tests/lib/subscriptionController.spec.ts:219]: nit: restore new line

**Review by Nana-EC — COMMENTED** [review]
Looking good, minor nits

**Review by AlfredoG87 — APPROVED** [review]

**Review by Nana-EC — DISMISSED** [review]

**Review by Nana-EC — APPROVED** [review]

**Pull request merged**